flexget/plugins/sites/alpharatio.py

```diff
@@ -0,0 +1,216 @@
+
+CATEGORIES = {
+    'tvsd': 'filter_cat[1]',
+    'tvhd': 'filter_cat[2]',
+    'tvdvdrip': 'filter_cat[3]',
+    'tvpacksd': 'filter_cat[4]',
+    'tvpackhd': 'filter_cat[5]',
+    'moviesd': 'filter_cat[6]',
+    'moviehd': 'filter_cat[7]',
+    'moviepacksd': 'filter_cat[8]',
+    'moviepackhd': 'filter_cat[9]',
+    'moviexxx': 'filter_cat[10]',
+    'mvid': 'filter_cat[11]',
+    'gamespc': 'filter_cat[12]',
+    'gamesxbox': 'filter_cat[13]',
+    'gamesps3': 'filter_cat[14]',
+    'gameswii':  'filter_cat[15]',
+    'appspc': 'filter_cat[16]',
+    'appsmac': 'filter_cat[17]',
+    'appslinux': 'filter_cat[18]',
+    'appsmobile': 'filter_cat[19]',
+    '0dayXXX': 'filter_cat[20]',
+    'ebook': 'filter_cat[21]',
+    'audiobook': 'filter_cat[22]',
+    'music': 'filter_cat[23]',
+    'misc': 'filter_cat[24]'
+}
+
+LEECHSTATUS = {
+    'normal': 0,
+    'freeleech': 1,
+    'neutral leech': 2,
+    'either': 3
+}
+
+
+class AlphaRatioCookie(Base):
+    __tablename__ = 'alpharatio_cookie'
+
+    username = Column(Unicode, primary_key=True)
+    _cookie = Column('cookie', Unicode)
+    cookie = json_synonym('_cookie')
+    expires = Column(DateTime)
+
+
+class SearchAlphaRatio(object):
+    """
+        AlphaRatio search plugin.
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+            'category': one_or_more({'type': 'string', 'enum': list(CATEGORIES.keys())}, unique_items=True),
+            'order_by': {'type': 'string', 'enum': ['seeders', 'leechers', 'time', 'size', 'year', 'snatched'],
+                         'default': 'time'},
+            'order_way': {'type': 'string', 'enum': ['desc', 'asc'], 'default': 'desc'},
+            'scene':  {'type': 'boolean', 'enum':  [True, False]},
+            'leechstatus': {'type': 'string', 'enum': list(LEECHSTATUS.keys()), 'default': 'normal'},
+        },
+        'required': ['username', 'password'],
+        'additionalProperties': False
+    }
+
+    base_url = 'https://alpharatio.cc/'
+    errors = False
+
+    def get(self, url, params, username, password, force=False):
+        """
+        Wrapper to allow refreshing the cookie if it is invalid for some reason
+        :param url:
+        :param params:
+        :param username:
+        :param password:
+        :param force: flag used to refresh the cookie forcefully ie. forgo DB lookup
+        :return:
+        """
+        cookies = self.get_login_cookie(username, password, force=force)
+
+        response = requests.get(url, params=params, cookies=cookies)
+
+        if self.base_url + 'login.php' in response.url:
+            if self.errors:
+                raise plugin.PluginError('AlphaRatio login cookie is invalid. Login page received?')
+            self.errors = True
+            # try again
+            response = self.get(url, params, username, password, force=True)
+
```

Comment:
Should probably set `errors` to False in the `else` branch (change it in morethantv plugin as well). Otherwise it will fail if the cookie becomes invalid in twice in the same session.